lib/http/client.rb

```diff
@@ -64,8 +64,11 @@ def make_request(req, options)
       @state = :dirty
 
       @connection ||= HTTP::Connection.new(req, options)
-      @connection.send_request(req)
-      @connection.read_headers!
+
+      unless @connection.failed_proxy_connect?
```

Comment:
A failed proxy connection should be an error, rather than soft failing I think. It's the equivalent of a socket error in that you're saying to connect to X and it failed.

If the concern is around accessing the response for further debugging, we can do something like what `Net::HTTP` does and include `response` as part of the error.